build/building/context.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Build context
+description: Learn how to use the build context to access files from your Dockerfile
+keywords: build, buildx, buildkit, context, git, tarball, stdin
+---
+
+The [`docker build`](../../engine/reference/commandline/build.md) or
+[`docker buildx build`](../../engine/reference/commandline/buildx_build.md)
+commands build Docker images from a [Dockerfile](../../engine/reference/builder.md)
+and a "context".
+
+A build's context is the set of files located in the specified `PATH` or `URL`.
+The build process can refer to any of the files in the context. For example,
+your build can use a [`COPY` instruction](../../engine/reference/builder.md#copy)
+to reference a file in the context or a [`RUN --mount=type=bind` instruction](../../engine/reference/builder.md#run---mounttypebind)
+for better performance with [BuildKit](../buildkit/index.md). The build context
+is processed recursively. So, a `PATH` includes any subdirectories and the
+`URL` includes the repository and its submodules.
+
+## `PATH` context
+
+When building with [BuildKit](../buildkit/index.md), which sets up an
+"interactive" session, only requested files that are needed by the build from
+the client are sent over the entire build context. On the other hand, the
+legacy builder will send the entire context (recursively) to the daemon.
+
+To increase the build's performance, exclude files and directories by adding
+a [`.dockerignore` file](../../engine/reference/builder.md#dockerignore-file) to
+the context directory.
```

Comment:
The previous paragraph makes it sound like we don't need to use a `.dockerignore` file if we are using BuildKit, because only files used in the build are sent over to the daemon anyway. What role does the `.dockerignore` file play in this context?

Reply:
> What role does the `.dockerignore` file play in this context?

With this instruction:

```dockerfile
COPY . .
```

All files will be sent anyway (BuildKit or legacy). The `.dockerignore` file can be useful to exclude some of them without having multi COPY instructions in your Dockerfile for the files/folders you need.

Also with BuildKit if you don't have `COPY` commands in the stages you are building there is no build context sent at all. If you have `COPY foo .` only file in build context for that build would be `foo` whereas legacy builder still send the whole context.

Reply:
I've added more explanation on this page, let me know if it seems clear enough.